tempest-internal/src/main/kotlin/app/cash/tempest/internal/Codec.kt

```diff
@@ -117,8 +117,9 @@ internal class ReflectionCodec<A : Any, D : Any> private constructor(
 
     fun addPrefix(dbItem: DB): DB {
       val attributeValues = mapAttributeValue.toAttributeValues(dbItem).toMutableMap()
-      for ((attributeName, prefix) in attributePrefixes) {
+      for ((attributeName, prefix, allowEmpty) in attributePrefixes) {
         val attributeValue = attributeValues[attributeName]
+        if (attributeValue == null && allowEmpty) continue
```

Comment:
would it be useful for clarity to update this to have a symmetry with the non-null case like:

```
if (attributeValue == null && allowEmpty) {
  attributeValues[attributeName] = null
```